python/triton/language/semantic.py

```diff
@@ -761,8 +761,10 @@ def store(ptr: tl.tensor,
     elt_ty = ptr_ty.element_ty
     # treat bool* as tl.int8*
     if elt_ty == tl.int1:
-        elt_ty_ptr = tl.int8
-        ptr_ty = tl.pointer_type(elt_ty_ptr, ptr_ty.address_space)
+        # convert to bool first and then store as int8
+        val = cast(val, tl.int1, builder)
```

Comment:
why didn't cast to `elt_ty` in line 772 work?

Reply:
Seems to me that the logic here is that `elt_ty` should be set to `tl.int8`, previously `elt_ty=tl.int1`.

I'd like to see more comments in the code.

Reply:
> why didn't cast to `elt_ty` in line 772 work?

Previously 772 does cast `val` to `tl.int1`, but somehow the logic of storing `tl.int1` is wrong. I didn't dig deeper, but I guess another solution is to fix the store logic somewhere.

Reply:
I think `val = cast(val, tl.int1, builder)` is redundant as `val` is overwritten by line 772.

Reply:
I think it's fine? If `val` gets stored to `int1*`, then we do `(int8*)ptr = val.to(int1).to(int8)` I don't really remember why `(int1*)ptr = val.to(int1)` didn't work though. Will be worth revisiting once the new backend lands